core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2212,9 +2212,9 @@ private[spark] object Utils extends Logging {
    * incompatible. In environments where dynamic allocation is turned on by default,
    * the latter should override the former (SPARK-9092).
    */
-  def isDynamicAllocationEnabled(conf: SparkConf): Boolean = {
+  def isDynamicAllocationEnabled(conf: SparkConf, isLocal: Boolean): Boolean = {
```

Comment:
actually it doesn't look very useful to pass in a flag that always decides the value of the return type. It appears that `SparkContext` is the only place where we need to read the flag so let's leave it out of this method.